### DISCUSSION TITLE
[Feature] WebXR Support for Viewing 360 assets in VR

### DISCUSSION THREAD

**Discussion by grazy27** [immich-app/immich#14768] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

VR is becoming increasingly popular and widely adopted across various domains, from gaming and entertainment to productivity and creative applications. With the rise of affordable and accessible VR headsets, such as the Oculus Quest series, users are looking for seamless ways to view and interact with their media in immersive environments.

Immich is already an excellent tool for managing and organizing photos and videos. However, the ability to view 360-degree photos and videos directly in VR is something that would be extremely nice to have.  

This can be implemented in 2 ways: 
* On android: Then Immich app will be installed on a headset directly,  and would allow more integrated interaction, but this approach demands significant efforts for implementation and support
* On web, using WebXR: Add a button to open 'VR' mode. Much simpler way of achieving this, libraries like A-Star simplify this process substantially. 

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by DeltaTango69** [comment]
I would very much welcome that..

**Comment by andreymaznyak** [comment]
I implement for fun, 360 preview using native web xr feature here https://github.com/immich-app/immich/pull/17988

You need remind that web xr works only with https and you should follow this guide for development https://medium.com/@schezfaz/getting-started-with-webxr-development-for-the-oculus-browser-159339b2a7f3

Example video here https://youtu.be/eeLxnJTiCZk
